modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/BlobManagerComponent.java

```diff
@@ -62,89 +55,36 @@
      */
     public static final String TRANSIENT_ID_PREFIX = "transient";
 
-    protected BlobProviderDescriptorRegistry blobProviderDescriptorsRegistry = new BlobProviderDescriptorRegistry();
-
-    protected Map<String, BlobProvider> blobProviders = new HashMap<>();
-
-    protected static class BlobProviderDescriptorRegistry extends SimpleContributionRegistry<BlobProviderDescriptor> {
-
-        @Override
-        public String getContributionId(BlobProviderDescriptor contrib) {
-            return contrib.name;
-        }
-
-        @Override
-        public BlobProviderDescriptor clone(BlobProviderDescriptor orig) {
-            return new BlobProviderDescriptor(orig);
-        }
-
-        @Override
-        public void merge(BlobProviderDescriptor src, BlobProviderDescriptor dst) {
-            dst.merge(src);
-        }
-
-        @Override
-        public boolean isSupportingMerge() {
-            return true;
-        }
-
-        public void clear() {
-            currentContribs.clear();
-        }
+    // volatile for double-checked locking
+    protected volatile boolean blobProvidersInitialized;
```

Comment:
I'm not seeing any use of this field for double-checked locking.

Reply:
Actually it's used in a synchronized method, so i can just remove "volatile" ?

Reply:
You can remove `volatile` but _all_ uses must be synchronized, even when you just set back to `false`:

    synchronized (this) {
        blobProvidersInitialized = false;
    }

Or keep `volatile` and use actual double-checked locking in `getBlobProviders`.

Reply:
:+1: (removed volatile)